sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -291,6 +291,8 @@ class RocksDB(
     acquire(LoadStore)
     recordedMetrics = None
     logInfo(log"Loading ${MDC(LogKeys.VERSION_NUM, version)}")
+    // Updating the current version, so the mapping can correctly identify reusable files.
+    rocksDBFileMapping.currentVersion = version
```

Comment:
Also, we set currentVersion to latestSnapshotVersion in the path of re-loading from DFS. With changelog checkpointing, rocksDBFileMapping.currentVersion would not be the loadedVersion, but would be the latestSnapshotVersion.
(I don't see the place to update the currentVersion, but please let me know if we somehow update it in load(). I know it will be updated in the load of "next" version.)

This does not seem to be the same with the above semantic where we follow with loadedVersion.

Reply:
@HeartSaVioR I understand your confusion. This `currentVersion` field in the mapping was something I didn't like and wanted to remove in the original code, but left it for another time since I had already done a lot of refactoring in that PR.

Please take a look at the code now, I have removed the use of `currentVersion` field. It wasn't meant to track the db loaded version. The code should be clearer now.

Reply:
I think it is much better, thanks!